### ISSUE TITLE
Strange connection issue

### ISSUE THREAD

**Issue by darookee** [EliasKotlyar/Xiaomi-Dafang-Hacks#436]
I [already mentioned this](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/336#issuecomment-387366241) a few weeks ago in another issue. I tried a few different things since then, but I sill got this problem so I decied to open another ticket.

I have two computers accessing the camera, my Home Assistant Server (192.168.0.1) and my PC (192.168.0.10). Everything works without a problem from my PC, the Web interface, playing the stream using VLC, connecting via ssh... But none of that works from my Server. Video and http/https worked for a few hours after I completely cleared the SD card and copied all the files again on monday, but now it is 'offline' again.

Ping works, but the camera is sending no response to my Server when it tries to connect via rtsp or http/https. I already checked the firewall on the server but that's not the problem. And it does not have any problems connecting to other devices in the network. And the realy strange thing is - MQTT works without a problem. The camera sends it's status message and receives the set-messages almost instantly.

Using `curl https://192.168.0.22/cgi-bin/currentpic.cgi` on the server results in a timeout and there is no entry in the logs on the camera.

**Comment by saipsa** [comment]
restart your router. I had similar issue, and restarting the router fixed it for me.

**Comment by darookee** [comment]
Already restarted the Accesspoint, Switch and Server and the Router, but no improvement... :-/

**Comment by darookee** [comment]
Over the weekend I again tried a few things to resolve this, without success:

- Reflash the camera and reinstall all the firmware files on a formatted SD card
- Give the computer which cannot connect another IP address and try to connect through this (to see if the camera blocks the IP)
- Boot the computer from an USB drive and use another IP for it (to see if the PC blocks the camera in some way)
- Plug the PC and the Accesspoint into other ports of the switch

All these without any success. The Computer just cannot 'speak' to the camera, but the camera can use it's MQTT server. I never encountered anything like this... :-|

**Comment by genehand** [comment]
Same thing, just got the camera and flashed it, everything was working great both from my web browser and pulling into home assistant with rtsp.  I can still ping it as well but can no longer connect to any ports.  The outbound MQTT connections are still fine here as well.  Restarted everything including my router for good measure, very odd..

**Comment by darookee** [comment]
@genehand thats good! Now we are at least two people with the same problem. Now it can be analyzed!
How are you running HA? On a Pi? In a container? Is it directly connected to your router?

**Comment by genehand** [comment]
Just moved to a NUC with Ubuntu 18.04 and installed Hass.io with docker, directly wired to my router.  Same issue from inside the container and the host OS.  Check this out though, I remembered that the camera feed seemed to be working in HA before I enabled MQTT.  So I disabled MQTT and rebooted the camera, and now I can pull the rtsp stream again.  It's not enough to stop the mqtt services, I had to reboot.  One step closer!

**Comment by darookee** [comment]
Wow! I just did the same and the stream and still image is working... 0_o
I also tried disabling only the MQTT controll or the MQTT status services. If either one is enabled my HA does get a connection to the camera, disabeling and rebooting fixes this again.

**Comment by fotiDim** [comment]
Same problem here. If MQTT controll or MQTT status are enabled Home Assistant loses times out:

`2018-07-23 23:46:44 WARNING (MainThread) [haffmpeg.tools] Timeout reading image.`

**Comment by binomie** [comment]
How much ram does your camera have? I had a dafang with a bootloader with 64 MB and had this issue, after applying #178 i got 128 MB and then i flashed the new bootloader.

Now the problems with Home assistant are gone. I have MQTT running and can access the images

**Comment by fotiDim** [comment]
@binomie that is what my camera reports when I do `cat /proc/cmdline `. Is that 128MB?
```
console=ttyS1,115200n8 mem=104M@0x0 ispmem=8M@0x6800000 rmem=16M@0x7000000 init=/linuxrc rootfstype=squashfs root=/dev/mtdblock2 rw mtdparts=jz_sfc:256k(boot),2048k(kernel),3392k(root),640k(driver),4736k(appfs),2048k(backupk),640k(backupd),2048k(backupa),256k(config),256k(para),-(flag)
```

**Comment by binomie** [comment]
Yes that is 128 MB version.
Try to install this bootloader:

https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/flashinguboot.md

But take care on the notices regarding bricking your camera

**Comment by darookee** [comment]
After updating to somewhere around 6d2c748 seemed to fix the rtsp connection problems (but why? 0_o).

I still cannot connect via ssh from my HA server, though, even after flashing the custom uboot. rtsp still works...

**Comment by fotiDim** [comment]
Updating to the latest version also fixed the problem for me. Perhaps this can be closed

**Comment by saipsa** [comment]
I have this issue on 1600x900 of connection drop happening sometimes on tiny cam, and all of the time on desktop vlc. 

connection is stable during night time. So I am guessing it has to do with the brightness of the scene. My DF3 captures outside.

I am using the custom bootloader, latest firmware. DF3 128MB

**Comment by bachya** [comment]
I seem to have this intermittently, as well (using 7221890c5a22): every so often, the feed remains visible via VLC, but Home Assistant gives the `Timeout reading image.` error.

@saipsa's comment is interesting, as my WyzeCam v2 gets a lot of direct sunlight. I'll see if it stabilizes at night.

**Comment by fotiDim** [comment]
Unfortunately I am back at the same problem. The only thing that I did is that I flashed the new boot loader to enable 1080p. However even at 720p the problem is there.

**Comment by humpedli** [comment]
Unfortunately I have the same issue, if I allow the MQTT status and control HA gets `Timeout reading image.` error. I can see the active feed on webui. If I disable those services and restart the camera, still image feed is working like a charm in HA.
My camera has only 64 MB RAM but I think it is enough. I also enabled SWAP, but it didn't help. I disabled every other service, there is always free memory and CPU usage is around 30%. I experienced this issue with multiple cameras.

**Comment by humpedli** [comment]
Update: 
If I disable MQTT status service and only enable the MQTT control service and restart the cameras, everything is working until the first published control message, after that there are no images anymore in HA. If I restart the cameras again, everything is working again.
I think somehow MQTT message can cause this problem and not the service itself and only one message is enough to mess up things.
@EliasKotlyar do you have any suggestion what should I try to investigate the issue? Thank you.

**Comment by tsvi** [comment]
Related issue.
While trying to debug the timeouts from the HASS side I ran ffmpeg manually with debug messaging. It looks like ffmpeg has a hard time trying to discover info about the sound stream.
It causes ffmpeg to try and collect 5sec of data from the stream to detect more info.
I tried manually stopping the h264 server and rerunning it with sound set to OFF. Suddenly running ffmpeg with time shows barely 1-2 sec to get a new picture.

I still get delays when sending mqtt commands. This is especially annoying when trying to turn left/right using mqtt.
